pitch.md

```diff
@@ -14,31 +18,82 @@ _What need is met by your idea? Who are you helping? What is that person's
 experience like now? What would their experience be like if you could help 
 them?_
 
+Often when writing robot code, quite a bit of work needs to go into writing the
+boiler plate for a state machine or interrupts or whatever structure is needed
+to get the actual logic implemented. It would be really nice for both
+inexperienced robot developers and people trying to write robot code quickly to
+be able to just specify the logic, and let someone else take care of the boiler
+plate.
+
+I've observed freshmen in E11 (autonomous vehicles) trying to write C++ for
+their robots. They neither know C++ very well nor have a great idea about how to
+write good robot code. A DSL can help solve both of these issues. The syntax
+could be somewhat nicer than C++'s (not a hard task), and the language can help
+steer programmers into using good design constructs by making them easy. In the
+C++ that E11 students are using, it's very easy to make a busy loop to wait for
+a certain amount of time and much more difficult to set up a state machine that
+will let their robot actually succeed. If they got to use this DSL, writing such
+a busy loop would be very difficult and the state machine would be simple,
+shoehorning students into using good robot design practices.
 
 ### Why a language?
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+A DSL is appropriate because the programmer wants to just specify logic,
+something programming languages are very good at. An internal DSL is probably
+appropriate so that calling out to the back-end functions is easy and users
+aren't constrained in any way by the language. An external DSL may also work,
+but in this case it might even compile to something like C++ because I want the
+language to be fast and very low overhead so that it can be run on many
+different systems (especially arduino).
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+This idea is really exciting to me. I have written such boiler plate several
+times and while it does make me feel like a HaXXor (to use Prof. O'Neil's
+phrasing) it's not the best use of my time, and it would be great if people who
+were good at logic but not so much at low-level programming could program
+awesome robots.
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain? Be careful
 to distinguish between, e.g,. a graphical interface and a linguistic interface._ 
 
+The syntax will be text-based and will look a lot like a standard GPL such as
+C++ or python. This is the right way because very complex logic might need to be
+implemented and writing a good graphical interface for something like this is
+extremely difficult. This language isn't trying to be super new-programmer
+friendly, it's just trying to make it easier for people to get to specifying
```

Comment:
So then the goal of the project is strictly to eliminate the boiler plate code? Do you think you might add constructs that don't exist in a current imperative language to make articulating robot logic easier? Maybe these new constructs aren't particularly "new-programmer friendly," but they _do_ aid in simplifying user expression.